File: lab2/compute.py
# CS326 Lab 2
# Compute integral of sin(x)

from math import sin

def integrate_sin(a,b,n):
   ''' Integrate sin(x) from a to b with n steps
   '''
   dx = (b-a)/n
   sum = 0
   for i in range(0,n):
      sum += sin(a+i*dx)
   return sum*dx

integral = integrate_sin(0, 3.14159, 10000000)
print(f'result = {integral}')

File: lab2/compute.c
/* CS326 Lab 2
   Author: D. Schuurman
   Compute integral of sin(x) 
*/

#include <math.h>
#include <stdio.h>

#define PI 3.14159
#define STEPS 10000000

// Integrate sin(x) from a to b with n steps
double integrate_sin(double a, double b, int n)
{
   int i;
   double dx = (b-a)/(double)n;
   double sum = 0;
   for (i=0; i<n; i++)
      sum += sin(a + i*dx);
   return sum*dx;
}

int main()
{
   double integral;
   integral = integrate_sin(0, PI, STEPS);
   printf("result = %.16f\n", integral);
   return 0;
}
